### ISSUE TITLE
Detect HTML inside format strings

### ISSUE THREAD

**Issue by paulschreiber** [WordPress/WordPress-Coding-Standards#1946]
## Bug Description
It should be helpful if WPCS detected HTML inside format strings (even if they aren't immediately printed).

## Minimal Code Snippet
```php
echo "<a href='" . $url . "'>" . $label . '</a>'; // escaping error detected
echo sprintf( '<a href="%s">%s</a>', $url, $label ); // escaping error detected
$boo = sprintf( '<a href="%s">%s</a>', $url, $label ); // escaping error NOT detected
```

## Error Code
WordPress.Security.EscapeOutput.OutputNotEscaped

## Environment

| Question               | Answer
| ------------------------| -------
| PHP version             | 7.3.11
| PHP_CodeSniffer version | 3.5.6
| WPCS version            | 2.3.0
| WPCS install type       | composer project local

## Tested Against `develop` branch?
- [ ] I have verified the issue still exists in the `develop` branch of WPCS.
